src/widget-editor/ReadMe.md

```diff
@@ -0,0 +1,45 @@
+# Admin App custom widget editor
+
+With this AdminApp extensibility point, you can create your own cusom widget designers and override the defualt auto generated designers.
+
+In this example we have two provided two samples:
+
+* For overriding a specific field for a widget designer.
+
+* For overriding the entire view for a widget designer.
+
+
+## Overriding a custom field
+
+The process is exactly the same as it is with overriding a field for a content type with the exception that the typeName of the item is equal to widget-{WIDGET_NAME}.
+
+The process invloves registering a custom field provider class by inheriting from the FieldsProvider provider interface and implementing the method "overrideField". This method is called for each and every field when it is beeing rendered in the interface. Thus you are able to replace each field.
+
+ The parameter that is passed to the method contains three properties:
+
+* fieldType -> The type of the field(shorttext, longtext, for all available field types see the FieldTypes class)
+* fieldName -> Holds the name of the field (Title, Content, Description)
+* typeName -> Holds the type, whose fields are currently beeing rendered.
+
+When a widget designer is opened, the typeName property will have the value of widget-{WIDGET_NAME}, where WIDGET_NAME is the name of the widget and you can choose to override the fields for each widget selectevely.
+
+For reference please refer to the [custom fields sample](../custom-fields/README.md) and the file "widget-editor-custom-fields-provider".
+
+
+## Overriding the entire designer
+
+If you wish to replace the appearance of the entire designer, then you can do so by inheriting from the WidgetEditorViewProvider interface. This interface gets called each time a widget designer is opened. It contains a single method(overrideView) which gets invoked by the AdminApp. This method gets invoked with a single parameter(context) that has the property widgetName. The name of the currently edited widget is passed as a value in this parameter. Thus you can selectevely deside whether to override the designer for a specific widget or not. For reference see the [CustomWidgetEditorViewProvider](custom-editor/custom-widget-editor-view-provider.ts) class.
```

Comment:
links to the API docs again (I suppose we would need to generate and update them)

Reply:
selectevely -> selectively

Reply:
deside -> decide